From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Wed, 9 May 2018 11:31:47 +0200
Subject: Avoid including the generated config-X11 private header

---
 libtaskmanager/virtualdesktopinfo.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libtaskmanager/virtualdesktopinfo.h b/libtaskmanager/virtualdesktopinfo.h
index aca6e76..5aac962 100644
--- a/libtaskmanager/virtualdesktopinfo.h
+++ b/libtaskmanager/virtualdesktopinfo.h
@@ -25,7 +25,8 @@ License along with this library.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "taskmanager_export.h"
 
-#include <config-X11.h>
+// private header
+// #include <config-X11.h>
 
 namespace TaskManager
 {
