---
 CMakeLists.txt |    2 ++
 1 file changed, 2 insertions(+)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,3 +270,5 @@ feature_summary(WHAT ALL INCLUDE_QUIET_P
 # add clang-format target for all our real source files
 file(GLOB_RECURSE ALL_CLANG_FORMAT_SOURCE_FILES *.cpp *.h)
 kde_clang_format(${ALL_CLANG_FORMAT_SOURCE_FILES})
+
+include(/usr/share/pkg-kde-tools/cmake/DebianABIManager.cmake)
