---
 CMakeLists.txt |    2 ++
 1 file changed, 2 insertions(+)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -411,3 +411,5 @@ kde_configure_git_pre_commit_hook(CHECKS
 file(GLOB_RECURSE ALL_CLANG_FORMAT_SOURCE_FILES *.cpp *.h)
 kde_clang_format(${ALL_CLANG_FORMAT_SOURCE_FILES})
 kde_configure_git_pre_commit_hook(CHECKS CLANG_FORMAT)
+
+include(/usr/share/pkg-kde-tools/cmake/DebianABIManager.cmake)
