diff --git a/applets/devicenotifier/package/contents/ui/DeviceItem.qml b/applets/devicenotifier/package/contents/ui/DeviceItem.qml
index d527ebe490e4d36546d5e96043fa176441da0c76..93569e20c10deef1dd8463d455e417990a1c42a3 100644
--- a/applets/devicenotifier/package/contents/ui/DeviceItem.qml
+++ b/applets/devicenotifier/package/contents/ui/DeviceItem.qml
@@ -72,29 +72,20 @@ PlasmaExtras.ExpandableListItem {
 
     // this keeps the delegate around for 5 seconds after the device has been
     // removed in case there was a message, such as "you can now safely remove this"
-    ListView.onRemove: {
-        deviceItem.isEnabled = false
-
-        if (deviceItem.hasMessage) {
-            ListView.delayRemove = true
-            keepDelegateTimer.restart()
-
-            ++devicenotifier.pendingDelegateRemoval // QTBUG-50380
-        }
+    ListView.onRemove: SequentialAnimation {
+        PropertyAction { target: deviceItem; property: "ListView.delayRemove"; value: deviceItem.hasMessage }
+        PropertyAction { target: deviceItem; property: "isEnabled"; value: false }
+        // Reset action model to hide the arrow
+        PropertyAction { target: deviceItem; property: "contextualActionsModel"; value: [] }
+        PropertyAction { target: deviceItem; property: "icon"; value: statusSource.lastIcon }
+        PropertyAction { target: deviceItem; property: "title"; value: statusSource.lastDescription }
+        PropertyAction { target: deviceItem; property: "subtitle"; value: statusSource.lastMessage }
+        PauseAnimation { duration: messageHighlightAnimator.duration }
+        // Otherwise the last message will show again when this device reappears
+        ScriptAction { script: statusSource.clearMessage(); }
         // Otherwise there are briefly multiple highlight effects
-        devicenotifier.currentIndex = -1
-    }
-
-    Timer {
-        id: keepDelegateTimer
-        interval: 5000 // same interval as the auto hide / passive timer
-        onTriggered: {
-            deviceItem.ListView.delayRemove = false
-            // otherwise the last message will show again when this device reappears
-            statusSource.clearMessage()
-
-            --devicenotifier.pendingDelegateRemoval // QTBUG-50380
-        }
+        PropertyAction { target: devicenotifier; property: "currentIndex"; value: -1 }
+        PropertyAction { target: deviceItem; property: "ListView.delayRemove"; value: false }
     }
 
     Timer {
diff --git a/applets/devicenotifier/package/contents/ui/FullRepresentation.qml b/applets/devicenotifier/package/contents/ui/FullRepresentation.qml
index 96db713da2abe9a28515597a33bdd1b0e4603cdf..11646183ac38c79a4ebaafac9eae624a1d20aa8a 100644
--- a/applets/devicenotifier/package/contents/ui/FullRepresentation.qml
+++ b/applets/devicenotifier/package/contents/ui/FullRepresentation.qml
@@ -79,6 +79,7 @@ PlasmaComponents3.Page {
             to: 0
             duration: PlasmaCore.Units.veryLongDuration * 8
             easing.type: Easing.InOutQuad
+            Component.onCompleted: devicenotifier.isMessageHighlightAnimatorRunning = Qt.binding(() => running);
         }
 
         Connections {
@@ -153,7 +154,7 @@ PlasmaComponents3.Page {
                 anchors.centerIn: parent
                 width: parent.width - (PlasmaCore.Units.largeSpacing * 4)
                 text: plasmoid.configuration.removableDevices ? i18n("No removable devices attached") : i18n("No disks available")
-                visible: notifierDialog.count === 0 && !devicenotifier.pendingDelegateRemoval
+                visible: notifierDialog.count === 0 && !messageHighlightAnimator.running
             }
         }
     }
diff --git a/applets/devicenotifier/package/contents/ui/devicenotifier.qml b/applets/devicenotifier/package/contents/ui/devicenotifier.qml
index 716fde7ae26e1e23215a73543ee14d19c9ba5aae..c6bd0e73f2f3f4cc164f3acc9e3a64940abaa3d2 100644
--- a/applets/devicenotifier/package/contents/ui/devicenotifier.qml
+++ b/applets/devicenotifier/package/contents/ui/devicenotifier.qml
@@ -39,7 +39,7 @@ Item {
     // QTBUG-50380: As soon as the item gets removed from the model, all of ListView's
     // properties (count, contentHeight) pretend the delegate doesn't exist anymore
     // causing our "No devices" heading to overlap with the remaining device
-    property int pendingDelegateRemoval: 0
+    property bool isMessageHighlightAnimatorRunning: false
 
     Plasmoid.switchWidth: PlasmaCore.Units.gridUnit * 10
     Plasmoid.switchHeight: PlasmaCore.Units.gridUnit * 10
@@ -64,7 +64,7 @@ Item {
         return "device-notifier"
     }
 
-    Plasmoid.status: (filterModel.count > 0 || pendingDelegateRemoval > 0) ? PlasmaCore.Types.ActiveStatus : PlasmaCore.Types.PassiveStatus
+    Plasmoid.status: (filterModel.count > 0 || isMessageHighlightAnimatorRunning) ? PlasmaCore.Types.ActiveStatus : PlasmaCore.Types.PassiveStatus
 
     PlasmaCore.DataSource {
         id: hpSource
@@ -189,6 +189,9 @@ Item {
         engine: "devicenotifications"
         property string last
         property string lastUdi
+        property string lastDescription
+        property string lastMessage
+        property string lastIcon
         onSourceAdded: {
             last = source;
             disconnectSource(source);
@@ -198,6 +201,9 @@ Item {
         onDataChanged: {
             if (last) {
                 lastUdi = data[last].udi
+                lastDescription = sdSource.data[lastUdi] ? sdSource.data[lastUdi].Description : ""
+                lastMessage = data[last].error
+                lastIcon = sdSource.data[lastUdi] ? sdSource.data[lastUdi].Icon : "device-notifier"
 
                 if (sdSource.isViableDevice(lastUdi)) {
                     plasmoid.expanded = true
@@ -209,6 +215,9 @@ Item {
         function clearMessage() {
             last = ""
             lastUdi = ""
+            lastDescription = ""
+            lastMessage = ""
+            lastIcon = ""
         }
     }
 
