From eaa5e74c04c56d035cfb307fc292f585ad530275 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Sun, 18 Aug 2019 09:51:40 +0200
Subject: [PATCH] Add missing include (QJsonArray) to fix compilation with Qt
 5.13

---
 dataengines/geolocation/location_ip.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dataengines/geolocation/location_ip.cpp b/dataengines/geolocation/location_ip.cpp
index e3abd36be..4a815ddf6 100644
--- a/dataengines/geolocation/location_ip.cpp
+++ b/dataengines/geolocation/location_ip.cpp
@@ -23,6 +23,7 @@
 #include <QDebug>
 #include <QJsonDocument>
 #include <QJsonObject>
+#include <QJsonArray>
 #include <QUrl>
 #include <KJob>
 #include <KIO/Job>
-- 
2.23.0

